massa-factory-worker/src/endorsement_factory.rs

```diff
@@ -193,10 +193,22 @@ impl EndorsementFactoryWorker {
 
         // store endorsements
         let mut endo_storage = self.channels.storage.clone_without_refs();
+        let endorsement_map_protocol = endorsements
+            .iter()
+            .map(|endo| (endo.id, endo.clone()))
+            .collect();
         endo_storage.store_endorsements(endorsements);
 
```

Comment:
would be nice to avoid cloning the endorsements, just to use `protocol.propagate_endorsmeents(endo_storage.clone())`

Reply:
@adrien-zinger FYI

Reply:
Yes I was thinking about it but as adrien is refactoring the protocol to use storage I didn't want to collide with him

Reply:
ok maybe let's merge this as is, but let's not forget about it. Can you add a comment inside Adrien's PR ?